### PR TITLE
CoilSensitivities.jl: Add realistic 2D coil profiles

### PR DESCRIPTION
**Pull request by scholand**

### CODE REVIEW

File: MRISimulation/src/CoilSensitivities.jl [hunk 1]
```diff
@@ -28,3 +29,112 @@ function birdcageSensitivity(N::Int64,
...
+        end
+    end
+    return val
+end
+
+
+function measured2DSensitivity(N::Int64,
+                               ncoils::Int64)
+
+    out = zeros(ComplexF64, N, N, 1, ncoils)
```
Comment (JeffFessler) [inline MRISimulation/src/CoilSensitivities.jl:134]:
```suggestion
    out = zeros(ComplexF32, N, N, 1, ncoils)
```
Do we need double precision for approximate coil maps?  Probably not.

Comment (JeffFessler) [reply MRISimulation/src/CoilSensitivities.jl:134]: Neat addition!  Do you really need `global` variables?

Comment (scholand) [reply MRISimulation/src/CoilSensitivities.jl:134]: > Do we need double precision for approximate coil maps? Probably not.

I agree. ComplexF32 should be fine, but shouldn't it be consistent with the output of birdcageSensitivity() in which out is a ComplexF64?

Comment (scholand) [reply MRISimulation/src/CoilSensitivities.jl:134]: > Neat addition! Do you really need `global` variables?

Then I will move the current global array in get_sens_coeff() and the integers in xsens.

Comment (JeffFessler) [reply MRISimulation/src/CoilSensitivities.jl]: I doubt that they must be consistent, and IMHO the birdcage code should also be modified to use F32!
An even better approach (for both functions) is to have an optional type `T` argument that defaults to `Float32` but allows the user to specify their preferred type.  I'll make that a code suggestion and you can decide.

File: MRISimulation/src/CoilSensitivities.jl [hunk 2]
```diff
@@ -28,3 +29,110 @@ function birdcageSensitivity(N::Int64,
...
+Checkout the original file:
+
+  https://github.com/mrirecon/bart/blob/master/src/simu/sens.c
+
+for more information.
+"""
+
+function get_sens_coeff(kx::Int64,
+                        ky::Int64,
+                        c::Int64)
```
Comment (JeffFessler) [inline MRISimulation/src/CoilSensitivities.jl:54]:
```suggestion
function get_sens_coeff(kx::Int,
                        ky::Int,
                        c::Int)
```
No reason to insist on 64 bit int is there?

File: MRISimulation/src/CoilSensitivities.jl [hunk 3]
```diff
@@ -28,3 +29,110 @@ function birdcageSensitivity(N::Int64,
...
+    @assert 8 >= c "Maximum number of supported coils was exceeded!"
+
+    cpos = [(pos[1] - dims[1] / 2) / (0.5 * dims[1]), (pos[2] - dims[2] / 2) / (0.5 * dims[2])]
+    val = 0 + 0im
+    coil_coeff = 5
+    shift = (coil_coeff - 1) / 2
+
+    for i in 1:coil_coeff
+        for j in 1:coil_coeff
+            val += get_sens_coeff(i, j, c) * exp(-2im * π * ((i - shift) * cpos[1] + (j - shift) * cpos[2]) / 4)
```
Comment (JeffFessler) [inline MRISimulation/src/CoilSensitivities.jl:122]:
```suggestion
            val += get_sens_coeff(i, j, c) * cis(-2 * π * ((i - shift) * cpos[1] + (j - shift) * cpos[2]) / 4)
```
supposedly `cis` is more efficient

File: MRISimulation/src/CoilSensitivities.jl [hunk 4]
```diff
@@ -28,3 +29,110 @@ function birdcageSensitivity(N::Int64,
...
+        end
+    end
+    return val
+end
+
+
+function measured2DSensitivity(N::Int64,
+                               ncoils::Int64)
+
+    out = zeros(ComplexF32, N, N, 1, ncoils)
```
Comment (JeffFessler) [inline MRISimulation/src/CoilSensitivities.jl:132]:
```suggestion
function measured2DSensitivity(N::Int64,
                               ncoils::Int64;
                               T::Type{<:Complex{<:AbstractFloat}} = ComplexF32)

    out = zeros(T, N, N, 1, ncoils)
```

Comment (JeffFessler) [reply MRISimulation/src/CoilSensitivities.jl:132]: i haven't tested this...

**Comment by JeffFessler** [comment]
i've enable the CI run so you can refine and see what passes the tests

File: MRISimulation/src/CoilSensitivities.jl (hunk 4 above)
Comment (scholand) [reply MRISimulation/src/CoilSensitivities.jl:132]: Thank you for all your suggestions! The latest one also works. The tests perform the same :)

**Comment by JeffFessler** [comment]
Ok, it looks good to me now.  @tknopp what do you think?

**Comment by tknopp** [comment]
looks good to me, thanks! Will merge but before releasing, we need to fix the CI errors.

**Pull request merged**